main/src/cgeo/geocaching/filters/core/DifficultyAndTerrainGeocacheFilter.java

```diff
@@ -0,0 +1,64 @@
+package cgeo.geocaching.filters.core;
+
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.storage.SqlBuilder;
+import cgeo.geocaching.utils.expressions.ExpressionConfig;
+
+public class DifficultyAndTerrainGeocacheFilter extends BaseGeocacheFilter {
+
+    public DifficultyGeocacheFilter difficultyGeocacheFilter = new DifficultyGeocacheFilter();
+    public TerrainGeocacheFilter terrainGeocacheFilter = new TerrainGeocacheFilter();
+
+    @Override
+    public Boolean filter(final Geocache cache) {
+        if (cache == null) {
+            return null;
+        }
+
+        final Boolean diffRange = difficultyGeocacheFilter.filter(cache);
+        final Boolean terrainRange = terrainGeocacheFilter.filter(cache);
+        if (diffRange == null) {
+            return terrainRange;
+        }
+        if (terrainRange == null) {
+            return diffRange;
+        }
+
+        return diffRange && terrainRange;
+    }
+
+    @Override
+    public boolean isFiltering() {
+        return difficultyGeocacheFilter.isFiltering() || terrainGeocacheFilter.isFiltering();
+    }
+
+    @Override
+    public void addToSql(final SqlBuilder sqlBuilder) {
+        difficultyGeocacheFilter.addToSql(sqlBuilder);
+        terrainGeocacheFilter.addToSql(sqlBuilder);
+    }
+
+    @Override
+    public void setConfig(final ExpressionConfig config) {
+        final ExpressionConfig diffConfig = new ExpressionConfig();
+        diffConfig.putDefaultList(config.get("difficulty"));
+        difficultyGeocacheFilter.setConfig(diffConfig);
+
```

Comment:
Don't know, if that is a good way to handle the config, probably there is a better / different way?

Reply:
Given that you reuse existing filters, this is a fine concept. Maybe a shortcut method can be added to "ExpressionConfig" to support this usage pattern - I never thought about the use case of a filter reusing other filters before :-))

Reply:
>  I never thought about the use case of a filter reusing other filters before :-))

It just came into my mind - and I thought just give it a try  :innocent: . If it does not work, I can still can create a independent one

Reply:
> Maybe a shortcut method can be added to "ExpressionConfig" to support this usage pattern 

Hm, do you mean a static method to create a new config from an existing one? Passing  the key and storing in default-list resp. other way round?

Reply:
Something like this:

in ExpressionConfig

```
/** creates a new config object, taking the content of one key of this config and putting it into the default list */
public void ExpressionsConfig getSubconfig(final String key) {
     final ExpressionConfig subConfig = new ExpressionConfig();
        subConfig.putDefaultList(this.get(key));
        return subConfig;
}
```

Reply:
Good idea